### PR TITLE
perf: measure torch.compile on the local embedder and fix the tokenizers pin that blocked it

### PR DESCRIPTION
**Pull request by Aureliolo**
Closes #2733.

**The issue asked for two things: measure, then gate behind a settings knob. The measurement was done first, as the issue required, and it says the knob should not be built. No knob ships here**; the benchmark, the numbers, and the reasoning do. That deviation is deliberate and is the subject of the "Why no knob" section below.

## What the measurement found

Model `all-MiniLM-L6-v2` (read from the shipped constant, not restated), batch size 1, `torch` 2.13.0, sentence-transformers 5.7.0, in a Linux container.

CPU, arms interleaved round-robin so machine drift cancels, 140 timed calls each:

| arm | median ms | p90 ms | speedup |
|---|---|---|---|
| eager | 11.357 | 13.557 | 1.00x |
| default, `dynamic=auto` | 9.382 | 11.714 | 1.21x |
| default, `dynamic=True` | 9.527 | 12.103 | 1.19x |

CUDA (RTX 4090), eager and candidate timed as drift-cancelled pairs:

| arm | eager ms | compiled ms | speedup | warm-up s | cosine |
|---|---|---|---|---|---|
| fp32 default, `auto` | 9.984 | 7.477 | 1.34x | 18.5 | 1.000000 |
| fp32 default, `dynamic=True` | 11.142 | 8.291 | 1.34x | 7.5 | 1.000000 |
| fp32 reduce-overhead, `dynamic=True` | 10.756 | 3.637 | **2.96x** | 10.6 | 1.000000 |
| bf16 reduce-overhead, `dynamic=True` | 9.510 | 3.872 | 2.46x | 9.6 | 0.999964 |
| bf16 default, `dynamic=True` | 11.359 | 8.220 | 1.38x | 8.8 | 0.999964 |

The issue's technical hypothesis is **confirmed**: `mode="reduce-overhead"` with `dynamic=True` reproduces the ~3x upstream reported, with vectors identical to eager. Two of its guesses are corrected: bf16 is not needed (compiled fp32 is both faster and leaves the vectors untouched), and the gain is a CUDA-graphs effect, so CPU only reaches ~1.2x. `dynamic=False` is the configuration to avoid: it compiles per shape until it trips the Dynamo recompile limit, then serves eager while still looking compiled.

## Why no knob

`EmbeddingSimilarityDetector` embeds one text per agent position, so a meeting performs 3 to 8 embed calls. On CPU, which is where this embedder runs, that saves 1.83 to 1.98 ms per call, so roughly 6 to 16 ms per meeting, against 10.8 to 25.5 s of one-off compilation: break-even needs about 5,100 to 13,700 calls in a single process lifetime, while a meeting already spends seconds per LLM call.

The 2.96x is real but out of reach here: it is a GPU effect, and no shipped deployment runs this embedder on a GPU. Three structural facts compound that, each verified rather than assumed, and each filed as a follow-up:

- The code path is **unreachable in the shipped runtime**. `api/auto_wire_meetings.py:249` builds the protocol registry once with a bare `StructuredPhasesConfig()`, and `MeetingProtocol.run(...)` takes no protocol config, so the per-meeting `embedder_strategy` never reaches the instance (#2738).
- The model is **hardcoded**, not operator-chosen, contrary to the issue's premise (#2739).
- The published backend image **ships no torch**, so the strategy raises there regardless (#2740).

A settings knob over that path would be a writable key that reaches nothing running, which is exactly the category `check_setting_live_or_compose_set.py` exists to abolish, and that gate has no per-line opt-out.

## Also fixed here: a live defect that blocked the measurement

`[tool.uv] override-dependencies` carried `tokenizers>=0.23.1`. A uv override replaces *every* declaration of a package, including the `tokenizers>=0.22.0,<=0.23.0` that `transformers` enforces at import through `require_version_core`. Since `tokenizers 0.23.0` was never published (only `0.23.0rc0`), `0.22.2` is the only satisfying release, so the lock installed `0.23.1` and `import sentence_transformers` raised `ImportError` wherever the fine-tune extra was installed. That broke the **entire Docker fine-tune pipeline** (mining, training, evaluation, preflight), with no resolver error to warn anyone.

It is a stale upper pin, not a real incompatibility: with the assertion bypassed, 0.23.1 loads models, tokenises and embeds correctly. Verified broken against the published image, and verified fixed by syncing the regenerated lock in a container and importing the package. The lock diff is a single version change; a Renovate hold stops the cap being re-proposed.

## What ships

- `scripts/measure_embedder_compile.py`: the benchmark, so the question can be re-answered after any torch or sentence-transformers bump, which is exactly when the answer changes. Arms are timed in drift-cancelled pairs, warm-up is reported rather than folded away, and each arm's output is checked against eager (an arm below the cosine floor exits non-zero rather than printing a number nobody reads).
- `tests/unit/scripts/test_measure_embedder_compile.py`: 44 tests over everything reachable without the ML extras, including all of `main` and both failure exits.
- `docs/reference/embedding-evaluation.md`: the tables, the method, and the verdict.

## Follow-ups filed

#2738 (dead protocol config), #2739 (hardcoded model), #2740 (no torch in the backend image), #2741 (fine-tune image has no C++ compiler and goes stale against `uv.lock`).

## Test plan

- `pytest tests/unit/memory -m unit`: 2102 passed (the issue's stated check).
- `pytest tests/unit/scripts/test_measure_embedder_compile.py`: 44 passed.
- Benchmark run end-to-end in a container: unknown arm exits 2, empty selection exits 2, a real run exits 0 and renders both formats.
- Dependency fix proven by `uv sync --frozen --group fine-tune-cpu` in a container followed by importing the package.
- ruff, ruff-format, mypy (full tree), vale, markdownlint, and all 70 convention gates clean.

## Review coverage

Pre-reviewed by 12 agents (0 failed); 31 findings triaged and all 31 implemented, including three verified numeric errors in the docs prose, an over-broad `ImportError` guard that would have misreported a missing triton, and partial results being discarded on a late failure.

No `src/synthorg/` behaviour changes, and no `api/` wiring touched, so nothing here collides with the sibling worktrees this wave.

### CODE REVIEW

File: scripts/measure_embedder_compile.py [hunk 1]
```diff
@@ -0,0 +1,669 @@
...
+
+def _timed(model: _EncodableModel, text: str, sync: Callable[[], None]) -> float:
+    """Return the wall-clock cost of one embed call, in milliseconds.
+
+    Returns:
+        The elapsed milliseconds.
+    """
+    started = time.perf_counter()
+    _encode(model, text, sync)
+    return (time.perf_counter() - started) * _MS_PER_SECOND
```
[the remark this answers is not in the document]
Comment (Aureliolo) [reply scripts/measure_embedder_compile.py:491]: Skipping this one: the premise does not hold against the current code.

`SentenceTransformerEmbedder.embed` **is** `encode()` plus the conversion, both inside the same method (`src/synthorg/memory/embedding/sentence_transformer.py:45-52`):

```python
def embed(self, text: str) -> tuple[float, ...]:
    vector = self._model.encode(text, normalize_embeddings=True)
    return tuple(vector.tolist())
```

So `tuple(vector.tolist())` is cost a caller of `embed()` pays on every call, not benchmark scaffolding layered on top of it. Timing it is what makes the published medians represent an embed call; dropping it from the timed region would make them understate what the meeting conflict detector actually spends, which is the quantity the "Why no setting was added" section reasons from. `_encode`'s docstring ("exactly as `SentenceTransformerEmbedder.embed` does") states that intent, and the current `_timed` meets it.

The observation that the constant falls on both arms is correct, and that is exactly why the per-call savings and the break-even arithmetic are unaffected either way; only the ratio would move, and it would move away from what a caller experiences.

The other five findings are fixed in this round, including the exit-code downgrade and its regression test.
